application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/wolframalpha/WolframAlphaCommandUtils.java

```diff
@@ -0,0 +1,226 @@
+                    "Your query is regarding The topic \"%s\" which might be supported by Wolfram Alpha in the future"
+                        .formatted(futureTopic.getTopic()));
+        }
+        LanguageMessage languageMsg = result.getLanguageMessage();
+        if (languageMsg != null) {
+            output.add(languageMsg.getEnglish() + "\n" + languageMsg.getOther());
+        }
+        DidYouMeans didYouMeans = result.getDidYouMeans();
+        if (didYouMeans != null && didYouMeans.getCount() != 0) {
+            if (didYouMeans.getCount() == 1)
+                output.add("Did you mean: " + didYouMeans.getDidYouMeans().get(0).getMessage());
+            else {
+                output.add("Did you mean \n" + didYouMeans.getDidYouMeans()
+                    .stream()
+                    .map(DidYouMean::getMessage)
+                    .map(text -> "• " + text)
+                    .collect(Collectors.joining("\n")));
+            }
+        }
+        RelatedExamples relatedExamples = result.getRelatedExamples();
+        if (relatedExamples != null && relatedExamples.getCount() != 0) {
+            if (relatedExamples.getCount() == 1) {
+                output.add("Here is a related example: "
+                        + relatedExamples.getRelatedExamples().get(0).getCategoryThumb());
+            } else {
+                output
+                    .add("Here are some related examples \n" + relatedExamples.getRelatedExamples()
+                        .stream()
+                        .map(RelatedExample::getCategoryThumb)
+                        .map(text -> "• " + text)
+                        .collect(Collectors.joining("\n")));
+            }
+        }
+        return String.join("\n", output);
+    }
+
+    static String handleError(QueryResult result) {
+        Error error = result.getErrorTag();
+        Constants.logger.error(
+                "Error getting response from Wolfram Alpha API: Error Code: {} Error Message: {}",
+                error.getCode(), error.getMessage());
+        return "An error occurred while getting response from the Wolfram|Alpha API. Check the URI";
+    }
+
+    static @NotNull Optional<QueryResult> parseQuery(@NotNull HttpResponse<String> response,
+            @NotNull WebhookMessageUpdateAction<Message> action) {
+        QueryResult result;
+        try {
+            Files.writeString(Path
+                .of("C:\\Users\\Abc\\IdeaProjects\\TJ-Bot-baseRepo\\application\\src\\main\\java\\org\\togetherjava\\tjbot\\commands\\mathcommands\\wolframalpha\\responsebody.xml"),
+                    response.body());
+            result = Constants.XML.readValue(response.body(), QueryResult.class);
+
+        } catch (IOException e) {
+            action.setContent("Unexpected response from WolframAlpha API").queue();
+            Constants.logger.error("Unable to deserialize the class ", e);
+            return Optional.empty();
+        }
+        return Optional.of(result);
```

Comment:
```suggestion
        }
        return Optional.ofNullable(result);
```

Subjective, so idk what your opinion is